### PR TITLE
[SPARK-5909][SQL] Add a clearCache command to Spark SQL's cache manager

### PR DESCRIPTION
**Pull request by yhuai**
JIRA: https://issues.apache.org/jira/browse/SPARK-5909

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSQLParser.scala [hunk 1]
```diff
@@ -74,9 +75,11 @@ private[sql] class SparkSQLParser(fallback: String => LogicalPlan) extends Abstr
     }
 
   private lazy val uncache: Parser[LogicalPlan] =
-    UNCACHE ~ TABLE ~> ident ^^ {
-      case tableName => UncacheTableCommand(tableName)
-    }
+    ( UNCACHE ~ TABLE ~> ident ^^ {
+        case tableName => UncacheTableCommand(tableName)
+      }
+      | CLEAR ~ CACHE ^^^ ClearCacheCommand
```
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSQLParser.scala]: Nit: don't indent this line

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala [hunk 2]
```diff
@@ -280,4 +280,20 @@ class CachedTableSuite extends QueryTest {
     assert(intercept[RuntimeException](table("t1")).getMessage.startsWith("Table Not Found"))
     assert(!isCached("t2"))
   }
+
+  test("Clear all cache") {
+    sql("SELECT key FROM testData LIMIT 10").registerTempTable("t1")
+    sql("SELECT key FROM testData LIMIT 5").registerTempTable("t2")
+    cacheTable("t1")
+    cacheTable("t1")
```
Comment (liancheng) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: Why do we cache `t1` twice?

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala [hunk 3]
```diff
@@ -280,4 +280,20 @@ class CachedTableSuite extends QueryTest {
...
+    clearCache()
+    assert(cacheManager.isEmpty)
+
+    sql("SELECT key FROM testData LIMIT 10").registerTempTable("t1")
+    sql("SELECT key FROM testData LIMIT 5").registerTempTable("t2")
+    cacheTable("t1")
+    cacheTable("t1")
```
Comment (liancheng) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: Also here.

**Comment by liancheng** [comment]
LGTM, merging to master, thanks!

**Comment by liancheng** [comment]
@marmbrus Should we backport this to branch-1.3?

**Comment by yhuai** [comment]
@liancheng Yes, please merge it to branch-1.3. Thanks!

**Comment by marmbrus** [comment]
I backported this.

**Pull request closed without merging**